BDArmory/UI/BDATargetManager.cs

```diff
@@ -589,6 +589,260 @@ private static float GetSeekerBias(float anglePos, float angleVel, FloatCurve se
             return seekerBias;
         }
 
+        public static float GetVesselAcousticSignature(Vessel v, TargetInfo ti, Vector3 sensorPosition = default(Vector3)) //not bothering with thermocline modelling at this time
+        {
+            float noiseScore = 1f;
+            float minNoise = float.MaxValue;
+            Part NoisePart = null;
+            bool hasEngines = false;
+            bool hasPumps = false;
+            hottestPart.Clear();
+            if (!v.Splashed) return 0;
+            var engineModules = VesselModuleRegistry.GetModules<ModuleEngines>(v);
+            if (engineModules != null)
+            {
+                if (engineModules.Count > 0)
+                {
+                    hasEngines = true;
+                    using (var engines = engineModules.GetEnumerator())
+                        while (engines.MoveNext())
+                        {
+                            if (engines.Current == null) continue;
+                            if (!engines.Current.EngineIgnited) continue;
+                            float thisScore = engines.Current.GetCurrentThrust() / 10; //pumps, fuel flow, cavitation, noise from ICE/turbine/etc.
+                            noiseScore = Mathf.Max(noiseScore, thisScore);
+                            minNoise = Mathf.Min(minNoise, thisScore);
+                            if (thisScore == noiseScore) NoisePart = engines.Current.part;
+                        }
+                }
+            }
+                var pumpModules = VesselModuleRegistry.GetModules<ModuleActiveRadiator>(v);
+                if (pumpModules != null)
+                {
+                    if (pumpModules.Count > 0)
```

Comment:
Found this was throwing NRE's somewhere; some debugging later and it's failing at the pumpModules.Count check;
VMR debug confirms that the radiatorModule is getting added to the registry, and returning a count of 0 on a test vessel sans radiators. The same checks for moduleEngines correctly work on engineless/engine-off vessels. Will look at it some more tomorrow when it's not late at night; in the meantime, you have any ideas?

Reply:
That doesn't really make any sense. `VMR.GetModules` always returns a `List` (which may be empty), so it should never fail accessing `.Count`.
Also, the `if (pumpModules != null)` and `if (engineModules != null)` checks are redundant, since they can never be `null`.

Reply:
Found and resolved this